cli/src/cmd/forge/script/receipts.rs

```diff
@@ -11,80 +11,62 @@ pub async fn wait_for_pending(
 ) -> eyre::Result<()> {
     if !deployment_sequence.pending.is_empty() {
         println!("##\nChecking previously pending transactions.");
-        let future_receipts = deployment_sequence
-            .pending
-            .iter()
-            .map(|tx_hash| pending_receipt(provider, *tx_hash))
-            .collect();
-        wait_for_receipts(future_receipts, deployment_sequence).await?;
+        wait_for_receipts(
+            &deployment_sequence.pending.iter().map(|tx_hash| Ok(*tx_hash)).collect::<Vec<_>>(),
+            deployment_sequence,
+            provider,
+        )
+        .await?;
     }
     Ok(())
 }
 
-/// Waits for a pending receipt, and gets its nonce to return (receipt, nonce).
-async fn pending_receipt(
-    provider: &Provider<Http>,
-    tx_hash: H256,
-) -> Result<(TransactionReceipt, U256), BroadcastError> {
-    let pending_err =
-        || BroadcastError::Simple(format!("Failed to get pending transaction {tx_hash:?}."));
-
-    let receipt = provider
-        .get_transaction_receipt(tx_hash)
-        .await
-        .map_err(|_| pending_err())?
-        .ok_or_else(pending_err)?;
-
-    let tx = provider
-        .get_transaction(tx_hash)
-        .await
-        .map_err(|_| pending_err())?
-        .ok_or_else(pending_err)?;
-
-    Ok((receipt, tx.nonce))
-}
-
 /// Waits for a list of receipts. If it fails, it tries to retrieve the transaction hash that can be
 /// used on a later run with `--resume`.
 pub async fn wait_for_receipts(
-    tasks: Vec<impl futures::Future<Output = Result<(TransactionReceipt, U256), BroadcastError>>>,
+    tx_hashes: &[Result<TxHash, BroadcastError>],
```

Comment:
is borrowing here necessary? because we clone the item later?

perhaps changes this to an impl IntoIter?